File: src/lightnap-ng/src/app/core/backend-api/dtos/users/request/search-claims-request-dto.ts
import { PagedRequestDto } from "../../paged-request-dto";

/**
 * Represents a request to search for claims with optional filtering and pagination.
 *
 * @extends PagedRequestDto
 *
 * @property {string} [type] - The username substring to search for.
 * @property {string} [value] - The value to search for in the claims.
 */
export interface SearchClaimsRequestDto extends PagedRequestDto {
  /**
   * The exact type to filter claims by.
   * @type {string}
   */
  type?: string;

  /**
   * The type substring to filter claims by.
   * @type {string}
   */
  typeContains?: string;

  /**
   * The exact value to filter claims by.
   * @type {string}
   */
  value?: string;

  /**
   * The value substring to filter claims by.
   * @type {string}
   */
  valueContains?: string;
}

File: src/LightNap.Core/Users/Dto/Request/SearchClaimsRequestDto.cs
using LightNap.Core.Api;

namespace LightNap.Core.Users.Dto.Request
{
    /// <summary>
    /// Represents a request to search claims.
    /// </summary>
    public class SearchClaimsRequestDto : PagedRequestDtoBase
    {
        /// <summary>
        /// Filter by exact claim type.
        /// </summary>
        public string? Type { get; set; }

        /// <summary>
        /// Filter by claim type substring.
        /// </summary>
        public string? TypeContains { get; set; }

        /// <summary>
        /// Filter by exact claim value.
        /// </summary>
        public string? Value { get; set; }

        /// <summary>
        /// Filter by claim value substring.
        /// </summary>
        public string? ValueContains { get; set; }
    }
}